internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -590,9 +594,12 @@ func (cf CloudFormation) errOnFailedStack(stackName string) error {
 	}
 	status := aws.StringValue(stack.StackStatus)
 	if cloudformation.StackStatus(status).IsFailure() {
+		events, _ := cf.cfnClient.ErrorEvents(stackName)
+
 		return &errFailedService{
 			stackName: stackName,
-			status: status,
+			service:   aws.StringValue(events[0].ResourceType),
```

Comment:
I'd like us to be safe here -
```
events, _ := cf.cfnClient.ErrorEvents(stackName)
var failedResourceType string
if events != nil {
   failedResourceType = events[0].ResourceType
}
return &errFailedService{
			stackName: stackName,
			service:   failedResourceType,
			status: status,
}
```

Just to avoid the potential index out of bound error!

Reply:
maybe this instead?
```
if len(events) > 0 {
  failedResourceType = events[0].ResourceType
}
```

https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices
> avoid making a distinction between a nil slice and a non-nil, zero-length slice, as this can lead to subtle programming errors.

Reply:
Implementing both these suggestions! :)